pkg/securitycontextconstraints/sccadmission/admission.go

```diff
@@ -384,14 +384,17 @@ func shouldIgnore(a admission.Attributes) (bool, error) {
 		return true, nil
 	}
 
-	_, ok := a.GetObject().(*coreapi.Pod)
+	pod, ok := a.GetObject().(*coreapi.Pod)
 	// if we can't convert then fail closed since we've already checked that this is supposed to be a pod object.
 	// this shouldn't normally happen during admission but could happen if an integrator passes a versioned
 	// pod object rather than an internal object.
 	if !ok {
 		return false, admission.NewForbidden(a, fmt.Errorf("object was marked as kind pod but was unable to be converted: %v", a.GetObject()))
 	}
-
+	// ignore all non Linux pods
+	if pod.Spec.OS != nil && pod.Spec.OS.Name != coreapi.Linux {
```

Comment:
Make it ignore Windows pods rather than ignoring all pods that are not Linux.

Is that safe? Who sets the OS field?

Reply:
> Make it ignore Windows pods rather than ignoring all pods that are not Linux.

I initially started with it. However we(upstream) thought of expanding OS field to have Solaris etc at a later stage. So to be future proof I have changed it to non-linux.

> Is that safe? Who sets the OS field?
The end user explicitly needs to set it. If not, it'll be defaulted to nil, so we should be good.

Reply:
> I initially started with it. However we(upstream) thought of expanding OS field to have Solaris etc at a later stage. So to be future proof I have changed it to non-linux.

I'd rather uncontrollably deny than uncontrollably allow.

What are the implications for the user on setting the field? Does that mean that the pod is _sure_ to schedule on a Windows node and _never_ on any other?

Reply:
> Does that mean that the pod is sure to schedule on a Windows node and never on any other?

There are no scheduling guarantees but if a Linux pod tries to run on Windows node, kubelet will reject it, so guarantees are at node level.

> I'd rather uncontrollably deny than uncontrollably allow.

I don't have a strong opinion on this, we can make it Windows for now and change it later if the need arises.

Reply:
> There are no scheduling guarantees but if a Linux pod tries to run on Windows node, kubelet will reject it, so guarantees are at node level.

Does it also work vice-versa, Windows pod will be rejected on a Linux system? I'm trying to prevent users from escaping the admission by simply setting the field. Is PSa turned off for Windows pods the same?